src/samples/doxygen/trampoline.cpp

```diff
@@ -0,0 +1,34 @@
+#include <rpp/rpp.hpp>
+#include <iostream>
+
+/**
+ * \example trampoline.cpp
+ **/
+int main()
+{
+    //! [trampoline]
+    rpp::source::just(rpp::schedulers::trampoline{}, 1, 2, 3)
+        .merge_with(rpp::source::just(rpp::schedulers::trampoline{}, 4, 5, 6))
+        .observe_on(rpp::schedulers::trampoline{})
+        .subscribe(
+            [](const int &v) { std::cout << "-" << v; },
+            [](const std::exception_ptr &error) {},
+            []() { std::cout << "-|" << std::endl; });
+    // Source 1: -1-2-3-|
+    // Source 2:        -4-5-6-|
+    // Output  : -1-2-3-4-5-6-|
+
+    // TODO: Enable this when we have "combine_latest" operator
+//    rpp::source::just(rpp::schedulers::trampoline{}, 1, 2, 3)
+//        .combine_latest(rpp::source::just(rpp::schedulers::trampoline{}, 1, 2, 3))
+//        .observe_on(rpp::schedulers::trampoline{})
+//        .subscribe(
+//            [](const std::tuple<int, int> &v) { std::cout << v << "-"; },
+//            [](const std::exception_ptr &error) {},
+//            []() { std::cout << "|" << std::endl; });
+//    // Source: -1-2-3--|
+//    //         -1-2-3--|
+//    // Output: {1,1}-{2,2}-{3,3}-|
```

Comment:
@victimsnino I found the implementation of `just()` with the scheduler is probably unexpected:

```
        worker.schedule([iterable=std::forward<decltype(iterable)>(iterable), 
                         subscriber=std::forward<decltype(subscriber)>(subscriber), 
                         index = size_t{0}]() mutable-> schedulers::optional_duration
        {            
            try
            {
                const auto& extracted_iterable = extract_iterable_from_packed(iterable);
                const auto  end = std::cend(extracted_iterable);
                auto        itr = std::cbegin(extracted_iterable);

                std::ranges::advance(itr, static_cast<int64_t>(index), end);

                if (itr != end)
                {
                    subscriber.on_next(utils::as_const(*itr));
                    if (std::next(itr) != end) // it was not last
                    {
                        ++index;
                        return schedulers::duration{}; // re-schedule this
                    }
                }

                subscriber.on_completed();
            }
            catch(...)
            {
                subscriber.on_error(std::current_exception());
            }
            return std::nullopt;

        });
```

So there'd be only one schedulable that emits the elements one by one via `on_next()`. And that will lead to the result of `-{3,1}-{3,2}-{3,3}-|`.

Reply:
Probably i didn't catch your point. Can't see if anything is wrong? We need to schedule each item emission and send only one value per scheduling. Expected result would be something like this
```
> inner observable send 1
> outer observable send 1
{1,1}
> inner observable send 2
{1,2}
>outer observable send 2
{2,2}
> inner observable send 3
{2, 3}
> outer observable send 3
{3,3}
```

Reply:
Oh I meant that I have this code and this output:

Code:
```
    rpp::source::just(rpp::schedulers::trampoline{}, 1, 2, 3)
        .with_latest_from(rpp::source::just(rpp::schedulers::trampoline{}, 4, 5, 6))
        .subscribe(
            [](const auto &v) { std::cout << v << "-"; },
            [](const std::exception_ptr &error) {},
            []() { std::cout << "|" << std::endl; });```

Output

```
-{1,6}-{2,6}-{3,6}-|
```

And the reason for that is because we only have one master schedulable that handles all the `on_next()`. Alternatively, we may consider creating new schedulable for each `on_next()`?

Reply:
In this case you need subscribe_on. I will fix after merging your PR due to it is required extra code in subscribe method

Reply:
Please let me know about the PR after this is merged.